include/redis3m/connection.h

```diff
@@ -27,6 +28,20 @@ class connection: utils::noncopyable
     typedef std::shared_ptr<connection> ptr_t;
 
     /**
+     * @brief Create and open a new connection with Timeout
+     * @param host hostname or ip of redis server, default localhost
+     * @param port port of redis server, default: 6379
+     * @param timeval timeout interval struct, default: 1 second
+     * @return
+     */
+    inline static ptr_t createTimeout(const std::string& host="localhost",
+                               const unsigned int port=6379,
+                               const struct timeval timeout=defaultTimeout())
```

Comment:
Yes, I can change to create_timeout.

I used struct timeval because that what is used by hiredis redisConnectWithTimeout
` struct timeval timeout = { 1, 500000 }; // 1.5 seconds
    c = redisConnectWithTimeout(hostname, port, timeout);`